src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.IO;
+using System.Collections.Specialized;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageServices
+{
+    public class PeekDefinition
+    {
+        private ConnectionInfo connectionInfo;
+        private string tempPath;
+
+        private Database database 
+        {
+            get
+            {
+                Server server = new Server(this.connectionInfo.SqlConnection.DataSource);
+                return server.Databases[this.connectionInfo.SqlConnection.Database];
+            }
+        }
+
+        public PeekDefinition(ConnectionInfo connInfo)
+        {
+            connectionInfo = connInfo;
+            tempPath = Path.GetTempPath();
+        }
+
+        /// <summary>
+        /// Script a table using SMO and write to a file.
+        /// </summary>
+        /// <param name="tableName">Table name</param>
+        /// <returns>Location object representing URI and range of the script file</returns>
+        internal Location[] GetTableDefinition(string tableName)
+        {
+            if (this.connectionInfo.SqlConnection != null)
+            {
+                Table table = database.Tables[tableName];
+                string tempFileName = String.Format("{0}{1}.sql", tempPath, tableName); 
+
+                if (table != null)
+                {
+                    using (StreamWriter scriptFile = new StreamWriter(File.Open(tempFileName, FileMode.Create, FileAccess.ReadWrite)))
+                    {
+                        StringCollection scripts = table.Script();
+                        foreach (string script in scripts)
+                        {
+                            if (script.IndexOf( "CREATE TABLE", StringComparison.OrdinalIgnoreCase) >= 0)
+                            {
+                                scriptFile.WriteLine(script);
+                            }  
+                        }
+
+                        
+                    }
+                    return GetLocationFromFile(tempFileName, 0);
+                }
+            }
+            return null;
```

Comment:
The code here is very duplicative for Table, View and Stored Procedure. We're going to have a lot more checks so we can simplify this dramatically. What's different is:
- SMO query to retries the object (table, view, proc)
- script.IndexOf check. The variable we check for is slightly different (basically string.format("CREATE {0}", typeName) should be used)

Based on this, please consider 1 common method that has the main logic, and takes as parameters 1) a getter for the SMO object and 2) The name to query for when trying to find the Create script. in scripts.